python/lsst/ctrl/bps/bps_reports.py

```diff
@@ -232,6 +232,52 @@ def __str__(self):
         return str("\n".join(lines))
 
 
+class ExitCodesReport(BaseRunReport):
+    """An extension of run report to give information about
+    error handling from the wms service.
+    """
+
+    def add(self, run_report, use_global_id=False):
+        # Docstring inherited from the base class.
+
+        # get labels from things and exit codes
+
+        labels = []
+        if run_report.run_summary:
+            for part in run_report.run_summary.split(";"):
+                label, count = part.split(":")
+                labels.append(label)
+        else:
+            id_ = run_report.global_wms_id if use_global_id else run_report.wms_id
+            self._msg = f"WARNING: Job summary for run '{id_}' not available, report maybe incomplete."
+            return
+        exit_code_summary = run_report.exit_code_summary
+        for label, exit_codes in zip(labels, exit_code_summary):
+            if exit_codes != [None]:
+                pipe_error_count = sum([code for code in exit_codes if code == 1])
```

Comment:
Maybe it would be worthwhile adding a comment that we're taking advantage of the fact that payloads always return 1 on failure here (right?).